lib/ui/src/Cardano/Wallet/UI/Deposit/Html/Common.hs

```diff
@@ -91,8 +95,13 @@ txIdH txId =
                 txId
 
 valueH :: Value -> Html ()
-valueH (ValueC (CoinC c) _) = do
-    span_ $ toHtml $ a ""
+valueH (ValueC (CoinC c) _) = lovelaceH $ fromIntegral c
+
+lovelaceH :: Natural -> Html ()
+lovelaceH c = do
+    span_ $ toHtml $ showLovelaceAsAda c
     span_ [class_ "opacity-25"] "₳"
-  where
-    a = showFFloatAlt @Double (Just 2) $ fromIntegral c / 1_000_000
+
+showLovelaceAsAda :: Integral a => a -> String
+showLovelaceAsAda c =
+    showFFloatAlt @Double (Just 2) (fromIntegral c / 1_000_000) ""
```

Comment:
It seems odd that I can enter `10.000001` as payment and not see the result:

```suggestion
    showFFloatAlt @Double (Just 6) (fromIntegral c / 1_000_000) ""
```

Reply:
Also, I wonder if the `Double` precision could be problematic...

Reply:
The maximum amount of Lovelace one can define is bounded to be a `Word64` if I am not mistaken, which should be representable as `Double` with some loss of precision. Do we want an exact representation? If so then this code is incorrect. If we don't really care which I think we don't because this UI will probably not go into production as is, then this is fine.